doc: Static Analysis requirement (#65207)
zephyrproject-rtos/zephyr: doc/contribute/guidelines.rst

you may select "Add me to project" to be added to the project. New members must
be approved by an admin.

Static analysis in the Zephyr codebase happens bi-weekly. GitHub issues are

> automatically created for any issues detected by Coverity. These issues will

Reviewer: ```suggestion
automatically created for any issues detected by the static analysis tool. These issues will
```

have the same (or equivalent) priority initially defined by the tool.

To ensure accountability and efficient issue resolution, they are assigned to

> the respective code owner who is responsible for the affected code.

Reviewer 2: Nit: I think "maintainer" was better than "codeowner" here.

Reviewer: +1

A dedicated team comprising members with expertise in static analysis, code

> quality, and software security is responsible to ensure the effectiveness of
> the static analysis process and verify that identified issues are properly

Reviewer 3: nit: grammar.
I suggest either:
```suggestion
quality, and software security ensure the effectiveness of
the static analysis process and verify that identified issues are properly
```
or
```suggestion
quality, and software security is responsible for ensuring the effectiveness of
the static analysis process and verifying that identified issues are properly
```

Reviewer: I like option 1.

Reviewer 4: Option 1 would be "ensures", not "ensure"

you may select "Add me to project" to be added to the project. New members must
be approved by an admin.

> Static analysis of the Zephyr codebase is conducted on a bi-weekly. GitHub

Reviewer 4: previous request for change was only partially incorporated leading to a missing word:

```suggestion
Static analysis of the Zephyr codebase is conducted on a bi-weekly basis. GitHub
```

issues are automatically created for any issues detected by static analysis
tools. These issues will have the same (or equivalent) priority initially
defined by the tool.

To ensure accountability and efficient issue resolution, they are assigned to
the respective maintainer who is responsible for the affected code.

A dedicated team comprising members with expertise in static analysis, code
quality, and software security ensures the effectiveness of the static

> analysis process and verify that identified issues are properly

Reviewer 4: ```suggestion
analysis process and verifies that identified issues are properly
```
